airbyte-integrations/connectors/source-amazon-seller-partner/source_amazon_seller_partner/streams.py

```diff
@@ -178,3 +239,28 @@ def parse_response(self, response: requests.Response, stream_state: Mapping[str,
         :return an iterable containing each record in the response
         """
         yield from response.json().get(self.data_field, {}).get(self.name, [])
+
+
+class VendorDirectFulfillmentShipping(AmazonSPStream):
+    name = "VendorDirectFulfillmentShipping"
+    primary_key = [["labelData", "packageIdentifier"]]
+    replication_start_date_field = "createdAfter"
+    next_page_token_field = "nextToken"
+    page_size_field = "limit"
+
+    def path(self, **kwargs) -> str:
+        return f"/vendor/directFulfillment/shipping/{VENDOR_API_VERSIONS}/shippingLabels"
+
+    def request_params(
+        self, stream_state: Mapping[str, Any], next_page_token: Mapping[str, Any] = None, **kwargs
+    ) -> MutableMapping[str, Any]:
+        params = super().request_params(stream_state, next_page_token, **kwargs)
+        if not next_page_token:
+            params.update({"createdBefore": pendulum.now("utc").strftime("%Y-%m-%dT%H:%M:%SZ")})
+        return params
+
+    def parse_response(self, response: requests.Response, stream_state: Mapping[str, Any], **kwargs) -> Iterable[Mapping]:
+        """
+        :return an iterable containing each record in the response
```

Comment:
it is quite obvious from the signature, 
more interesting can be the fact that we extract only shippingLabels